torch/distributed/distributed_c10d.py

```diff
@@ -149,7 +150,7 @@ def supports_complex(reduceOp: ReduceOp) -> bool:
     return reduceOp not in denyList
 
 
-class Backend:
+class Backend(str):
```

Comment:
Is this really a string subclass?

Reply:
Yes... and it always has just been a str factory.  It's a very weird class.  The `__new__` method actually just returned a string always.  AFAICT there has never even been an instance of one of these classes!  We had a bunch of Union types for some reason.  But what the code seemed to really be wanting to do _is_ properly done subclassing a built-in.  So I think it's okay here, and lets us later actually potentially create real instances of these classes.